shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -371,9 +371,31 @@ final class ImageReaderSurfaceProducer
     private int requestedWidth = 0;
     private int requestedHeight = 0;
 
+    private class PerImage {
```

Comment:
nit: Consider a more descriptive name. Please add a doc comment about what to use this class for.

Reply:
(and when to call close)